pipeline-runner/R/gem2s-7-upload_to_aws.R

```diff
@@ -149,7 +154,11 @@ meta_sets <- function(input, scdata, color_pool) {
 
     for (j in seq_along(values)) {
       value <- values[j]
-      cell_ids <- scdata$cells_id[scdata[[seurat_key]] == value]
+
+      cell_ids <- list()
+      for (scdata in scdata_list) {
+        cell_ids <- append(cell_ids,  scdata$cells_id[scdata[[seurat_key]] == value])
```

Comment:
what is `seurat_key`?

Reply:
it's the seurat representation of the metadata track name supplied by the user. Basically, it gets passed through the make.names() function, which converts strings to a syntactically valid form for R (for example:  suppose a user named one of their metadata tracks like: "1-non-valid-name". dashes are not allowed in variable names in R. And they can't start with numbers either. so `make.names("1-non-valid-name")` gives you `"X1.non.valid.name"`.

I renamed all these variables to make them clearer, and added docs, because it took me a second to understand what was going on.